### PR TITLE
Allow injection of HTTP transport to enable HTTP replayer pattern

### PR DESCRIPTION
**Pull request by tcesnik-veza**
## Changes
Allowing to inject HTTP transport of a client through configuration, to enable use cases involving HTTP replayers: https://github.com/google/go-replayers.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -105,6 +105,8 @@ type Config struct {
 	// Number of seconds to keep retrying HTTP requests. Default is 300 (5 minutes)
 	RetryTimeoutSeconds int `name:"retry_timeout_seconds" auth:"-"`
 
+	HTTPTransport *http.Transport
```
Comment (nfx) [inline config/config.go:108]:
```suggestion
	HTTPTransport http.RoundTripper
```

`*http.Transport` is too concrete in this place. can't do that.

**Review by nfx — CHANGES_REQUESTED** [review]

File: config/config.go (hunk 1 above)
Comment (nfx) [reply config/config.go:108]: @tcesnik-veza please also add a unit test for this change.

File: go.mod [hunk 2]
```diff
@@ -22,6 +22,7 @@ require (
 	github.com/google/s2a-go v0.1.7 // indirect
 	github.com/googleapis/enterprise-certificate-proxy v0.3.2 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
+	github.com/stretchr/objx v0.5.0 // indirect
```
Comment (nfx) [inline go.mod:25]: please implement the feature without any external dependencies. you don't need `mock.Mock` - you can just implement a dummy type that has a `RoundTrip(request *http.Request) (*http.Response, error)` just for unit tests.

**Review by nfx — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (tcesnik-veza) [reply go.mod:25]: Ok I can change this, although please note that the mock library (`github.com/stretchr/testify v1.8.4`) is already a part of the project, but for some reason its indirect dependency was not yet included. Let me know how would you like to proceed.

Comment (nfx) [reply go.mod:25]: simplest mock is 

```
type hc func(r *http.Request) (*http.Response, error)

func (cb hc) RoundTrip(r *http.Request) (*http.Response, error) {
	return cb(r)
}
```

**Review by nfx — APPROVED** [review]

**Review by mgyucht — APPROVED** [review]

**Comment by mgyucht** [comment]
@tcesnik-veza thank you for raising this! Can you please setup commit signing using a GPG key and make sure all your commits are signed? See https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits for guidelines.

**Pull request merged**